### PR TITLE
javascript-javascript3-week1

### PR DESCRIPTION
**Pull request by somayehdvr**

### CODE REVIEW

File: javascript/javascript3/redo/week1/giphyApi.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+    let limitGifNumber = document.querySelector('.limitGifNumber').value
+    if (!limitGifNumber) { limitGifNumber = 3 }
+
+    fetch(`https://api.giphy.com/v1/gifs/search?api_key=KV1LXsbBYQmH1VEYZqM65HZdwRt9U0fT&q=${gifWord}&limit=${limitGifNumber}&offset=0&rating=g&lang=en`)
+        .then(response => response.json())
+        .then(gifData => {
+            const gifDataDiv = document.querySelector(".gifDataDiv")
+            gifDataDiv.innerHTML = ``
+            for (let i = 0; i < limitGifNumber; i++) {
```
Comment (miltonhultgren) [inline javascript/javascript3/redo/week1/giphyApi.js:18]: What if the API responds with less images than the limit the user picked?

File: javascript/javascript3/redo/week1/weatherApp.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+const weatherDataDiv = document.querySelector(".weatherDataDiv")
+
+function fetchWeatherData() {
+    const city = document.querySelector(".city").value
+    if (!city) { city = "copenhagen" }
+
+    fetch(`https://api.openweathermap.org/data/2.5/weather?q=${city}&appid=85bf3bafaf784a301c66129107680db7`)
+        .then(response => response.json())
+        .then(weatherData => {
+            weatherDataDiv.innerHTML = `
```
Comment (miltonhultgren) [inline javascript/javascript3/redo/week1/weatherApp.js:11]: Just curious, do you feel comfortable with the DOM API as well?
Using innerHTML with template strings feels very mustache like so it's easier to write out but it's good to also understand the DOM API.

**Review by miltonhultgren — APPROVED** [review]
This looks good to me 👍🏼 

I like how you name your variables and functions, and that you extract functions for the main work! Also great that you account for values missing.

But I find your formatting style hard to read, I would look into using Prettier or the built in VSCode formatting tools. Your code makes me think of an essay style which is different from most coding styles. Coding is a lot about familiarity so there is a big value in following the common style for the language you're working in. I found your style to also remind me of Ruby, but Ruby has a lot of special keywords that allow it to be written more like natural language. JavaScript doesn't have those syntax features so it's usually formatted differently.

I wouldn't worry about or aim to keep my code as "small" as possible, it's better to make it as readable as possible.
Once tools like Webpack has packed it into a bundle it will be much, much smaller anyway!

File: javascript/javascript3/redo/week1/weatherApp.js (hunk 2 above)
Comment (somayehdvr) [reply javascript/javascript3/redo/week1/weatherApp.js:11]: Thanks for mentioning. I read more about it.

**Comment by somayehdvr** [comment]
Thank you for the comment. Yes, I'm using Prettier. But I will try to write more extended to be easier to read.

**Pull request merged**